### PR TITLE
Consider rating overrides for `NEW_VULNERABILITY` and `NEW_VULNERABLE_DEPENDENCY` notifications

### PR DESCRIPTION
**Pull request by nscuro**
### Description

Updates `NEW_VULNERABILITY` and `NEW_VULNERABLE_DEPENDENCY` notifications to include rating overrides instead of the original vulnerability ratings, if overrides are present.

Also reduces code duplication across `RowMapper` implementations.

> [!WARNING]
> ~This PR builds on top of #474, I'll rebase it once the other PR is merged. The change set will be a lot smaller then. :)~

### Addressed Issue

Closes https://github.com/DependencyTrack/hyades/issues/967

### Additional Details

N/A

### Checklist

- [x] I have read and understand the [contributing guidelines](../CONTRIBUTING.md#pull-requests)
- [x] This PR fixes a defect, and I have provided tests to verify that the fix is effective
- ~This PR implements an enhancement, and I have provided tests to verify that it works as intended~
- ~This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)~
- ~This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

File: src/main/java/org/dependencytrack/policy/cel/persistence/CelPolicyVulnerabilityRowMapper.java [hunk 1]
```diff
@@ -85,44 +76,32 @@ public Vulnerability map(final ResultSet rs, final StatementContext ctx) throws
...
+        return stringArray(rs, columnName).stream()
+                .map(Integer::parseInt)
+                .toList();
     }
 
-    private void maybeSetCwes(final ResultSet rs, final Consumer<Iterable<Integer>> setter) throws SQLException {
-        if (!hasColumn(rs, "cwes")) {
-            return;
+    private static List<Vulnerability.Alias> maybeConvertAliases(final ResultSet rs, final String columnName) throws SQLException {
+        final List<VulnerabilityAlias> aliases = json(rs, columnName, VULNERABILITY_ALIASES_TYPE_REF);
```
Comment (mehab) [inline src/main/java/org/dependencytrack/policy/cel/persistence/CelPolicyVulnerabilityRowMapper.java:86]: can we rename this method to say convertJsonToGivenType or something?

File: src/main/java/org/dependencytrack/persistence/jdbi/mapping/RowMapperUtil.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            setter.accept(value);
+        }
+    }
+
+    public static boolean hasColumn(final ResultSet rs, final String columnName) throws SQLException {
+        try {
+            return rs.findColumn(columnName) >= 0;
+        } catch (SQLException e) {
+            if (e instanceof final PSQLException pe) {
+                if (PSQLState.UNDEFINED_COLUMN.getState().equals(pe.getSQLState())) {
```
Comment (mehab) [inline src/main/java/org/dependencytrack/persistence/jdbi/mapping/RowMapperUtil.java:66]: is there going to be an else case here of return true or another exception needs to be added

File: src/main/java/org/dependencytrack/policy/cel/persistence/CelPolicyVulnerabilityRowMapper.java (hunk 1 above)
Comment (nscuro) [reply src/main/java/org/dependencytrack/policy/cel/persistence/CelPolicyVulnerabilityRowMapper.java]: Renamed to `deserializeJson` to make it more clear what the method is doing.

**Review by mehab — APPROVED** [review]
lgtm

**Pull request merged**